src/Documentation/2.0/Configuration2.0.md

```diff
@@ -103,15 +105,30 @@ var builder = new ClientBuilder()
     .UseConfiguration(config)
     // Add assemblies to scan for grains interfaces and serializers.
     // For more info read the Application Parts section
-    .AddApplicationPartsFromReferences(typeof(IHello).Assembly)
+    .ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(IHello).Assembly))
     .ConfigureLogging(logging => logging.AddConsole())
 var client = builder.Build();
 await client.Connect();
 ```
 
 ## Application Parts
 
-// TODO
+Orleans 2.0 does not perform automatic folder scanning to discover user assemblies and types. Instead, those assemblies are provided explicitly during the configuration stage. These assemblies are refered to as Application Parts. All Grains, Grain Interfaces, and Serializers are discovered using Application Parts.
+
+Application Parts are configured using an `IApplicationPartsManager`, which can be accessed using the `ConfigureApplicationParts` extension method on `IClientBuilder` and `ISiloHostBuilder`. The `ConfigureApplicationParts` method accepts a delegate, `Action<IApplicationPartManager>`.
+
+The following extension methods on `IApplicationPartManager` support common uses:
+* `AddApplicationPart(assembly)` adds a single assembly can be added using the extension method.
+* `AddFromAppDomain()` adds all assemblies in the current `AppDomain`.
```

Comment:
That are currently loaded. It's important to mention that assemblies that get loaded afterwards will not be considered